pkg/registry/core/service/rest.go

```diff
@@ -432,19 +435,55 @@ func (rs *REST) ResourceLocation(ctx genericapirequest.Context, id string) (*url
 		}
 		for i := range ss.Ports {
 			if ss.Ports[i].Name == portStr {
-				// Pick a random address.
-				ip := ss.Addresses[rand.Intn(len(ss.Addresses))].IP
-				port := int(ss.Ports[i].Port)
-				return &url.URL{
-					Scheme: svcScheme,
-					Host:   net.JoinHostPort(ip, strconv.Itoa(port)),
-				}, rs.proxyTransport, nil
+                                // This is a little wonky, but it's expensive to test for the presence of a Pod
```

Comment:
can pick a random starting point and iterate, wrapping around the end of the list, like `ssSeed` and `ss := &eps.Subsets[(ssSeed+ssi)%len(eps.Subsets)]` does above